### PR TITLE
Some Docker improvements

### PR DESCRIPTION
**Pull request by Wonderfall**
- improve docker_entrypoint.sh
- serve static files with puma by default (so it won't break for people using Docker Hub)
- sort packages list
- use virtual package for build deps
- show how to assign UID/GID

### CODE REVIEW

File: docker_entrypoint.sh [hunk 1]
```diff
@@ -1,4 +1,14 @@
 #!/bin/sh
+
+### 1. Adds local user (UID and GID are provided from environment variables).
+### 2. Updates permissions, except for ./public/system (should be chown on previous installations).
```
Comment (Gargron) [inline docker_entrypoint.sh]: Is there any reason why you can't chown that here, too?

Comment (Wonderfall) [inline docker_entrypoint.sh]: It would take way too long even with my optimised command. I can try with GNU parallel but this seems a bit hacky IMO... 🤔

**Comment by Wonderfall** [comment]
Hmm please wait a bit before merging, I got an error with `redis-rack` while downloading.
That said it's not due to these changes (I'm 99% it's working but I prefer to test before).

File: docker_entrypoint.sh (hunk 1 above)
Comment (Gargron) [inline docker_entrypoint.sh]: Okay, makes sense.

**Comment by Gargron** [comment]
@Wonderfall Okay. I think this is the last bit before 1.4 though, so the sooner this is ready the sooner v1.4rc1 can be tagged

**Comment by Wonderfall** [comment]
It can be merged now, it works.

**Review by Gargron — APPROVED** [review]

**Pull request merged**